change curve algorihm for rating graph to curveCatmullRom to fix visual glitches

Kry-Vosa opened the pull request:
`curveCatmullRom` was chosen over `curveMonotoneX`.
`curveMonotoneX` give better looking curve, however it introduces a minor visual glitch where the "middle of uncertainty" skill line doesn't follow the edge of "bottom part of confidence interval" and leaves a little gap, see image.
If we wanna use the nicer looking `curveMonotoneX`, we could also omit the "middle of uncertainty" skill line, since it is barely visible anyway and the edge of the two areas accomplishes the same visual feat.

![gap](https://github.com/user-attachments/assets/86f29dac-a725-483a-b3c2-d04ea7ce4962)

L-e-x-o-n commented:
What does `curveCatmullRom` look like?

NortySpock commented:
Not to open a can of worms, but:  Why a curve vs using a straight line segment, like a "normal" line chart?

Kry-Vosa commented:
`curveCatmullRom` would look like this:
![image](https://github.com/user-attachments/assets/12cd2c93-5daf-4816-a06e-5ba6cee7d8c0)

NortySpock commented:
To me, the artificial curves looked wrong, like someone was trying to paint their own lines on a chart just to hide the fact that their OS sucked. ("See, right there, just for a moment I was totally above 17OS!") 

I prefer line segments, but then I am far too literal and matter-of-fact about things. Also I am unfun at parties, and am trending below 10 OS. 😛 🙃

L-e-x-o-n commented:
I prefer the flatness of `curveMonotoneX` and it undershooting less.
The uncertainty gap is not as bad and the line could be removed entirely as you said.
In any case, all of these options fix the very obvious visual glitches of the previous curve and are an improvement.

Kry-Vosa commented:
@NortySpock `curveMonotoneX` preserves local minima and maxima (which includes global minima and maxima), meaning the curve never overshoots (or undershoots) on the Y axis.

Kry-Vosa commented:
Updated the commit - switched to `curveMonotoneX`, removed the "middle of uncertainty" line. This is how it looks with in the updated pull request: 
![image](https://github.com/user-attachments/assets/590208bd-4b51-44ff-b318-7eee46b6294f)
